### PR TITLE
Replace deprecated functions.

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**
Deprecated functions might get removed in future Python versions. We should avoid there usage. This PR activates the corresponding pylint warning and uses the newer replacements for deprecated functions where possible.

**Interactions with other PRs:**
none, I believe

**How has this been tested?**
tests still pass

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- Bug fix (non-breaking change which fixes an issue)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [n/a] I have updated the documentation accordingly.
- [n/a] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Seanny123** [comment]
The changes look good to me, but cause `flake8` errors, because the comments make the lines too long. So I added a quick commit to fix that. Let me know if you think it's acceptable to move those `pylint-disable` comments.

File: nengo/ipynb.py [hunk 1]
```diff
@@ -14,7 +14,8 @@
     from html import escape
 except ImportError:
     from cgi import escape as cgi_escape
-    escape = lambda s, quote=True: cgi_escape(s, quote=quote)  # pylint: disable=deprecated-method
+    # pylint: disable=deprecated-method
```
Comment (jgosmann) [inline nengo/ipynb.py]: This might disable the warning for the whole file, but I have to double check.

**Comment by jgosmann** [comment]
`pylint` also has a line length check that supports a regex for ignoring certain lines. That way it is for example possible to ignore overlong URLs without a `noqa` in front. I would enable the pylint check and disable the flake8 check.

This is not to say that it might not be better to fix the overlong lines in this PR. Those are somewhat ugly, but fixing it also changes the scoping of the disabling (the check could be reenabled after the line, but then you get two lines of clutter which is also annoying).

File: nengo/ipynb.py (hunk 1 above)
Comment (jgosmann) [inline nengo/ipynb.py]: It just applies to the block, so I think this is ok.

**Comment by jgosmann** [comment]
Double checking with the [documentation](https://docs.pylint.org/en/latest/user_guide/message-control.html), I think the amendments are fine. :)

**Comment by Seanny123** [comment]
I fixed a `pylint` error that I missed in my last commit. Now the static check is still failing on TravisCI, but for reasons I've already resolved in #1298 which will be resolved when this is rebased.

**Comment by jgosmann** [comment]
I think that the failure might be related to this PR. The AppVeyor failure is unrelated, though.

**Pull request merged**